src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx

```diff
@@ -586,15 +586,15 @@ export default class CreateSingleItemModal extends React.PureComponent<Props, St
     const { weareblePreviewUpdated, type, model } = this.state
     // if model is an image, the wearable preview won't be needed
     if (model && isImageFile(model)) {
-      this.getMetricsAndScreenshot()
+      await this.getMetricsAndScreenshot()
       return
     }
 
     const controller = WearablePreview.createController('thumbnail-picker')
     this.setState({ previewController: controller })
 
     if (weareblePreviewUpdated) {
-      this.getMetricsAndScreenshot()
+      await this.getMetricsAndScreenshot()
```

Comment:
Was this await forgotten or was it placed on purpose to prevent blocking the UI?

Reply:
I think it was initially forgotten because this step occurs when loading the model for the first time